src/libraries/Microsoft.Extensions.DependencyInjection.Abstractions/src/ActivatorUtilities.cs

```diff
@@ -212,12 +247,11 @@ private static MethodInfo GetMethodInfo<T>(Expression<T> expr)
             if (!TryFindPreferredConstructor(instanceType, argumentTypes, ref constructorInfo, ref parameterMap) &&
                 !TryFindMatchingConstructor(instanceType, argumentTypes, ref constructorInfo, ref parameterMap))
             {
-                string? message = $"A suitable constructor for type '{instanceType}' could not be located. Ensure the type is concrete and all parameters of a public constructor are either registered as services or passed as arguments. Also ensure no extraneous arguments are provided.";
-                throw new InvalidOperationException(message);
+                throw new InvalidOperationException(SR.Format(SR.CtorNotLocated, instanceType));
             }
 
-            matchingConstructor = constructorInfo;
-            matchingParameterMap = parameterMap;
+            matchingConstructor = constructorInfo!;
```

Comment:
Are these `!` necessary? Both `TryFindPreferredConstructor` and `TryFindMatchingConstructor` are annotated as they won't return `null` parameters when returning `true`. So the compiler should be able to see that these will not be null.

Reply:
I thought it would add to readability